internal/dataplane/translator/backendref.go

```diff
@@ -28,46 +30,57 @@ func backendRefsToKongStateBackends(
 			if backendRef.Port != nil {
 				port = int32(*backendRef.Port)
 			}
-			backend := kongstate.ServiceBackend{
-				Name: string(backendRef.Name),
-				PortDef: kongstate.PortDef{
+			namespace := route.GetNamespace()
+			if backendRef.Namespace != nil {
+				namespace = string(*backendRef.Namespace)
+			}
+			backend, err := kongstate.NewServiceBackendForService(
+				namespace,
+				string(backendRef.Name),
+				kongstate.PortDef{
 					Mode:   kongstate.PortModeByNumber,
 					Number: port,
 				},
-				Weight: backendRef.Weight,
+			)
+			if err != nil {
+				logger.Error(err, "failed to create ServiceBackend for backendRef")
 			}
-			if backendRef.Namespace != nil {
-				backend.Namespace = string(*backendRef.Namespace)
+			if backendRef.Weight != nil {
+				backend.SetWeight(*backendRef.Weight)
 			}
 			backends = append(backends, backend)
 		} else {
 			// we log impermissible refs rather than failing the entire rule. while we cannot actually route to
 			// these, we do not want a single impermissible ref to take the entire rule offline. in the case of edits,
 			// failing the entire rule could potentially delete routes that were previously online and in use, and
 			// that remain viable (because they still have some permissible backendRefs)
-			var (
-				namespace = route.GetNamespace()
-				kind      string
-			)
-			if backendRef.Namespace != nil {
-				namespace = string(*backendRef.Namespace)
-			}
-			if backendRef.Kind != nil {
-				kind = string(*backendRef.Kind)
-			}
-
-			objName := fmt.Sprintf("%s %s/%s",
-				route.GetObjectKind().GroupVersionKind().String(),
-				route.GetNamespace(),
-				route.GetName())
-			logger.Error(nil, "Object requested backendRef to target, but no ReferenceGrant permits it, skipping...",
-				"object_name", objName,
-				"target_kind", kind,
-				"target_namespace", namespace,
-				"target_name", backendRef.Name,
-			)
+			logger.Error(nil, "Object requested backendRef to target, but no ReferenceGrant permits it, skipping...")
 		}
 	}
 
 	return backends
 }
+
+func loggerForBackendRef(logger logr.Logger, route client.Object, backendRef gatewayapi.BackendRef) logr.Logger {
+	var (
+		namespace = route.GetNamespace()
+		kind      string
```

Comment:
```suggestion
		kind      string = "unknown"
```
maybe or something similar "unspecified" to not have an empty field in log message, but an indication that it wasn't set. WDYT?